src/CabanaPD_Boundary.hpp

```diff
@@ -304,12 +375,59 @@ auto createBoundaryCondition( UserFunctor user_functor, ExecSpace exec_space,
 {
     using memory_space = typename Particles::memory_space;
     using bc_index_type = BoundaryIndexSpace<memory_space, BoundaryType>;
-    bc_index_type bc_indices = createBoundaryIndexSpace<BoundaryType>(
+    bc_index_type bc_indices = createBoundaryIndexSpace(
         exec_space, particles, planes, initial_guess );
     return BoundaryCondition<bc_index_type, UserFunctor>(
         bc_indices, user_functor, force_update );
 }
 
+// Custom index space cases
+template <class BCTag, class ExecSpace, class BoundaryParticles>
+auto createBoundaryCondition( BCTag, const double value,
+                              BoundaryParticles particles )
+{
+    using memory_space = typename BoundaryParticles::memory_space;
+    using bc_index_type = BoundaryIndexSpace<memory_space, Custom>;
+    bc_index_type bc_indices = createBoundaryIndexSpace( particles );
+    return BoundaryCondition<bc_index_type, BCTag>( value, bc_indices );
+}
+
+template <class UserFunctor, class BoundaryParticles>
+auto createBoundaryCondition( UserFunctor user_functor,
+                              BoundaryParticles particles,
+                              const bool force_update )
+{
+    using memory_space = typename BoundaryParticles::memory_space;
+    using bc_index_type = BoundaryIndexSpace<memory_space, Custom>;
+    bc_index_type bc_indices = createBoundaryIndexSpace( particles );
+    return BoundaryCondition<bc_index_type, UserFunctor>(
+        bc_indices, user_functor, force_update );
+}
+
+// Wrappers for single plane cases.
+template <class BoundaryType, class BCTag, class ExecSpace, class Particles>
+auto createBoundaryCondition( BCTag tag, const double value,
```

Comment:
Why do we need two "createBoundaryCondition" (Lines 409 and 419)?

Reply:
They have different inputs (tagged versions vs custom)